azavea/temperate
# Updates climate-change-components and Angular CLI dependencies

## Overview

We modified climate-change-components to use peer dependencies instead of normal dependencies out of a desire to reduce the number of duplicated dependencies in this project.

To that end I also updated the Angular CLI to version 1.4.10, which fixes a recurring issue we've encountered with an incorrect peer dependency of `@angular-devkit/schematics`.

With these changes in place, I was able to flatten RxJS from 3 versions to 1 and core-js from 2 versions to 1 🎉

### Demo

N/A

### Notes

Removing duplicates from `yarn.lock` is kind of annoying, and requires doing a specific set of steps
 1) Find the what is causing the duplicate versions (we had dependencies that had to be upgraded, dependencies that should have been using `peerDependencies` but weren't, and a version of `rxjs` in our `package.json` that was too restrictive)
 2) Fix those problems in the `package.json`
 3) Remove _all_ versions of the duplicated package from `yarn.lock`
 4) Shell into the Angular container and run `yarn install`
 5) Verify in the `yarn.lock` that the package in question is no longer duplicated


## Testing Instructions

 * `scripts/update`
 * `scripts/server`
 * Check that indicator charts are still working correctly

Closes azavea/climate-change-components#30
